apps/full-site-editing/full-site-editing-plugin/premium-content/view.js

```diff
@@ -1,44 +1,49 @@
-		if ( uri.match( re ) ) {
-			return uri.replace( re, '$1' + key + '=' + value + '$2' );
+let premiumContentJWTToken = '';
+
+/**
+ *
+ * @param {event} eventFromIframe - message event that gets emmited in the checkout iframe.
+ * @listens message
+ */
+function handleIframeResult( eventFromIframe ) {
+	if ( eventFromIframe.origin === 'https://subscribe.wordpress.com' && eventFromIframe.data ) {
+		const data = JSON.parse( eventFromIframe.data );
+		if ( data && data.result && data.result.jwt_token ) {
+			// We save the token for now, doing nothing.
+			premiumContentJWTToken = data.result.jwt_token;
+			// We will also set this in a cookie  - just in case. This will be reloaded in the refresh, when user clicks OK.
+			// But user can close the browser window before clicking OK. IN that case, we want to leave a cookie behind.
+			const date = new Date();
+			date.setTime( date.getTime() + 365 * 24 * 60 * 60 * 1000 );
+			document.cookie =
+				'jp-premium-content-session' +
+				'=' +
+				premiumContentJWTToken +
+				'; expires=' +
+				date.toGMTString() +
+				'; path=/';
+		}
+		if ( data && data.action === 'close' && premiumContentJWTToken ) {
+			document.location.href = updateQueryStringParameter(
+				document.location.href,
+				'token',
+				premiumContentJWTToken
+			);
 		}
-		return uri + separator + key + '=' + value;
 	}
-	let premiumContentJWTToken = '';
+}
 
-	window.addEventListener(
-		'message',
-		function handleIframeResult( eventFromIframe ) {
-			if ( eventFromIframe.origin === 'https://subscribe.wordpress.com' && eventFromIframe.data ) {
-				const data = JSON.parse( eventFromIframe.data );
-				if ( data && data.result && data.result.jwt_token ) {
-					// We save the token for now, doing nothing.
-					premiumContentJWTToken = data.result.jwt_token;
-					// We will also set this in a cookie  - just in case. This will be reloaded in the refresh, when user clicks OK.
-					// But user can close the browser window before clicking OK. IN that case, we want to leave a cookie behind.
-					const date = new Date();
-					date.setTime( date.getTime() + 365 * 24 * 60 * 60 * 1000 );
-					document.cookie =
-						'jp-premium-content-session' +
-						'=' +
-						premiumContentJWTToken +
-						'; expires=' +
-						date.toGMTString() +
-						'; path=/';
-				}
-				if ( data && data.action === 'close' && premiumContentJWTToken ) {
-					document.location.href = updateQueryStringParameter(
-						document.location.href,
-						'token',
-						premiumContentJWTToken
-					);
-				}
-			}
-		},
-		false
-	);
-} );
+function updateQueryStringParameter( uri, key, value ) {
+	const re = new RegExp( '([?&])' + key + '=.*?(&|$)', 'i' );
+	const separator = uri.indexOf( '?' ) !== -1 ? '&' : '?';
+	if ( uri.match( re ) ) {
+		return uri.replace( re, '$1' + key + '=' + value + '$2' );
+	}
+	return uri + separator + key + '=' + value;
+}
+
+document.addEventListener('DOMContentLoaded', (event) => {
```

Comment:
I would avoid fat arrow. This is not transpiled so this may not always work on every browser
We could also move the function declaration of updateQueryStringParameter and handleIframeResult into the callback to avoid bleeding into globals

Reply:
Good idea, I'll make those changes.